### PR TITLE
Implement GraphQL query and subscription for secrets

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- add query and subscription for secrets

**Related issue(s)**

See also https://github.com/kyma-project/console/issues/623

### CODE REVIEW

File: components/console-backend-service/internal/domain/k8s/secret_resolver.go [hunk 1]
```diff
@@ -3,39 +3,91 @@ package k8s
...
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-func newSecretResolver(secretGetter v1.SecretsGetter) *secretResolver {
+//go:generate mockery -name=secretSvc -output=automock -outpkg=automock -case=underscore
+type secretSvc interface {
+	Find(name, namespace string) (*v1.Secret, error)
+	List(namespace string, params pager.PagingParams) ([]*v1.Secret, error)
+	//Update(name, namespace string, update v1.Secret) (*v1.Secret, error)
```
Comment (pkosiec) [inline components/console-backend-service/internal/domain/k8s/secret_resolver.go]: Please remove comments

File: components/console-backend-service/internal/domain/k8s/secret_service_test.go [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+	t.Run("Nil", func(t *testing.T) {
+		secretInformer, _ := fixSecretInformer()
+		svc := k8s.NewSecretService(secretInformer, nil)
+
+		svc.Unsubscribe(nil)
+	})
+}
+
+//
```
Comment (pkosiec) [inline components/console-backend-service/internal/domain/k8s/secret_service_test.go]: Remove comments or introduce mtation in the same PR

File: components/console-backend-service/internal/gqlschema/schema.graphql [hunk 3]
```diff
@@ -709,6 +718,8 @@ type Query {
 
     backendModules: [BackendModule!]!
 
+    secrets(namespace: String!, first: Int, offset: Int): [Secret!]!
```
Comment (pkosiec) [inline components/console-backend-service/internal/gqlschema/schema.graphql]: Define `@HasAccess` directive

File: components/console-backend-service/internal/gqlschema/schema.graphql [hunk 4]
```diff
@@ -752,6 +763,7 @@ type Subscription {
     applicationEvent: ApplicationEvent!,
     podEvent(namespace: String!): PodEvent!
     serviceEvent(namespace: String!): ServiceEvent! @HasAccess(attributes: {resource: "services", verb: "watch", apiGroup: "", apiVersion: "v1", namespaceArg: "namespace"})
+    secretEvent(namespace: String!): SecretEvent!
```
Comment (pkosiec) [inline components/console-backend-service/internal/gqlschema/schema.graphql]: Define `@HasAccess` directive

File: components/console-backend-service/internal/gqlschema/schema.graphql (hunk 3 above)
Comment (pkosiec) [inline components/console-backend-service/internal/gqlschema/schema.graphql]: What about query for single resource, similar to `pod`?

File: tests/console-backend-service/internal/domain/k8s/secret_test.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+	Data         string    `json:"data"`
+	Type         string    `json:"type"`
+	CreationTime time.Time `json:"creationTime"`
+	Labels       string    `json:"labels"`
+	Annotations  string    `json:"annotations"`
+}
+
+type secretStatusType string
+
+func TestSecret(t *testing.T) {
```
Comment (pkosiec) [inline tests/console-backend-service/internal/domain/k8s/secret_test.go]: Hmm, but the test will always fail as there is no secret created

**Comment by aerfio** [comment]
`tests/console-backend-service/internal/domain/k8s/secret_test.go` will be created in new PR with mutations 👍

File: resources/core/charts/console-backend-service/templates/cluster-role.yaml [hunk 6]
```diff
@@ -1,52 +1,52 @@
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1beta1
 metadata:
-  name: {{ template "fullname" . }}
-  labels:
-    app: {{ template "name" . }}
+    name: {{ template "fullname" . }}
+    labels:
+        app: {{ template "name" . }}
```
Comment (pkosiec) [inline resources/core/charts/console-backend-service/templates/cluster-role.yaml]: Please reset formatting of the file. Thanks!

**Comment by aerfio** [comment]
Closed in favor of #3312

**Pull request closed without merging**